### DISCUSSION TITLE
Processing IMU without GPS?

### DISCUSSION THREAD

**Discussion by kvoit** [rodralez/NaveGo#150] [Q&A] [answered]
The README suggests that "Processing of an inertial navigation system (INS)." is possible, which I read as generating attitude and position (relative to a starting point) from only INS data. In the examples, I only seem to find code with GNSS, though.

In particular, we have 100Hz 3-axis acceleration/gyro/magnetometer data and want to reconstruct attitude and position in post processing, for segments of few minutes, within an area of around 20m radius. Pitch and roll would be by far more important than yaw and position, though.

Is this the right tool? Which examples would I have to look at or am I overlooking some crucial documentation besides the examples?

**Comment by rodralez** [comment, marked as answer]
Hello kvoit,

>The README suggests that "Processing of an inertial navigation system (INS)." is possible, which I read as generating attitude and position (relative to a starting point) from only INS data. In the examples, I only seem to find code with GNSS, though.

Typically, an INS works find only if a high-end IMU is used (~ USD 100k). This is how airplanes navigated when GPS didn't exist. 

The common approach is to use an INS/GNSS system where a low-cost IMU is involved. This is why most code that you will find is based on this system.

>In particular, we have 100Hz 3-axis acceleration/gyro/magnetometer data and want to reconstruct attitude and position in post processing, for segments of few minutes, within an area of around 20m radius. Pitch and roll would be by far more important than yaw and position, though.

>Is this the right tool? Which examples would I have to look at or am I overlooking some crucial documentation besides the examples?

Yes, you can use NaveGo only using a INS. You have to modify file [` ins_gnss.m` ](https://github.com/rodralez/NaveGo/blob/master/ins-gnss/ins_gnss.m ). You have to delete lines from 317 to 435 and delete some extra variables.   

Please, let us know how your experiments develop!

Kind regards.

**Reply by Mtulaza1** [reply]
Hello, I have struggle using mentioned method to delete lines from 317 to 435 to obtain INS only simulation using a data from IMU.  Could you please add a file with deletes variables to simulate only INS? 
Kind regards.

**Reply by rodralez** [reply]
Hello Mtulaza1,

I created the new issue #151 to address your request. Please, be patient. I will try to process it ASAP.

Kind regards.

**Reply by Mtulaza1** [reply]
I would be grateful, thanks for your respond.

**Reply by rodralez** [reply]
The INS-only example is ready. 

Please, download the [development branch](https://github.com/rodralez/NaveGo/tree/develop), not the master branch.

Run the [navego_example_real_ins_ekinox.m](https://github.com/rodralez/NaveGo/blob/develop/examples/real-data/navego_example_real_ins_ekinox.m).

As you will see, the performance of this system is really bad.

Regards.
